File: SPOJ/HACKRNDM.py
def bin(ele,start,end):
	if ele > list1[end] or ele < list1[start]:
		return 0
	while (end > start):
		mid = (start + end)/2
		if list1[mid] == ele:
			return 1
		elif list1[mid] > ele:
			end = mid
		else:
			start = mid + 1
	return 0
n,k = map(int, raw_input().split())
list1 = map(int, raw_input().split())
list1.sort()
count = 0
for i in xrange(n):
	if bin(list1[i] + k,i+1,n-1):	
		count += 1

print count

File: SPOJ/HACKRNDM.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
	int n,k;
	scanf("%d %d",&n,&k);
	int arr1[n],count = 0;
	unordered_map<int,int> ele;
	for(int i = 0;i<n;i++){
		scanf("%d",&arr1[i]);
		ele[arr1[i]] = 1;
	}
	sort(arr1,arr1 + n);
	for(int i = 0;i<n;i++){
		if (ele.find(arr1[i]+k) != ele.end()){
			count += 1;
		}
	}
	printf("%d",count);
	return 0;
}